src/app/dashboard/accounts/page.tsx

```diff
@@ -28,26 +51,94 @@ export default function AccountsPage(): JSX.Element {
   accounts = accounts || [];
   todayPrices = todayPrices || new PriceDBMap();
 
+  let tree: AccountsTree = {
+    account: {},
+    total: new Money(0, 'EUR'),
+    children: [],
+  };
+  const root = accounts.find(a => a.type === 'ROOT');
+  if (root && !todayPrices.isEmpty) {
+    tree = buildNestedRows(root, accounts, todayPrices, dateRange);
+  }
+
   return (
     <>
-      <div className="grid grid-cols-12 items-center pb-4">
-        <span className="col-span-10 text-xl font-medium">
-          Accounts
+      <div className="flex items-center pb-4">
+        <span className="text-xl font-medium">
+          Your finances
         </span>
-        <div className="col-span-2 col-end-13 justify-self-end">
-          <AddAccountButton
-            onSave={() => {
-              save();
-              mutate('/api/accounts');
-              mutate('/api/accounts/splits');
-            }}
+        <span className="ml-auto mr-3">
+          <DateRangeInput
+            earliestDate={earliestDate}
+            dateRange={dateRange}
+            onChange={setDateRange}
           />
+        </span>
+        <div>
+          <AddAccountButton />
+        </div>
+      </div>
+      <div className="grid grid-cols-12 items-top pb-4">
+        <div className="col-span-3 p-4 mr-4 rounded-sm bg-gunmetal-700">
+          <AccountsTable tree={tree} />
         </div>
       </div>
-      <AccountsTable
-        accounts={accounts}
-        todayPrices={todayPrices}
-      />
     </>
   );
 }
+
+/**
+ * This function recursively goes through all the accounts so it can build the tree
+ * of accounts containing total and subrows so they can be rendered in react table.
+ *
+ * Note that it interacts with PriceDB so prices can be aggregated to the parent's
+ * total in order to display total amounts of assets, income, etc.
+ */
+function buildNestedRows(
+  current: Account,
+  accounts: Account[],
+  todayQuotes: PriceDBMap,
+  dateRange: Interval,
+): AccountsTree {
+  const { childrenIds } = current;
+  let total = current.getTotal(dateRange);
+  let subRows: AccountsTree[] = [];
+
+  if (current.type === 'ROOT') {
```

Comment:
Hmmm to try in the future but we could actually compute the total net worth in the Root account. Use mainCurrency as currency.